### PR TITLE
samples: Bluetooth: df: Fix possible stack overflow issue

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Host receive thread has small default stack size.
Attempt to convert data from binany to string requires
large array. Allocation of such array on stack could
cause stack overflow.

To limit the scope of the array and avoid stack overflow
the array is changed to be static local variable in a function.

Also the array has been prepared to accept max advertising
data sieze of 1650 bytes.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/direction_finding_connectionless_rx/src/main.c [hunk 1]
```diff
@@ -147,8 +147,8 @@ static void recv_cb(struct bt_le_per_adv_sync *sync,
 		    const struct bt_le_per_adv_sync_recv_info *info,
 		    struct net_buf_simple *buf)
 {
+	static char data_str[BT_GAP_ADV_MAX_EXT_ADV_DATA_LEN];
```
Comment (alwa-nordic) [inline samples/bluetooth/direction_finding_connectionless_rx/src/main.c:150]: The hexadecimal representation is two chars per byte plus null-terminator. I think this should be:
```suggestion
	static char data_str[BT_GAP_ADV_MAX_EXT_ADV_DATA_LEN * 2 + 1];
```

**Review by alwa-nordic — APPROVED** [review]

File: samples/bluetooth/direction_finding_connectionless_rx/src/main.c (hunk 1 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_connectionless_rx/src/main.c]: I do not understand, could you rephrase your question?

**Review by Thalley — APPROVED** [review]
